ProcessLib/HeatTransportBHE/BHE/BHE_1U.cpp

```diff
@@ -0,0 +1,258 @@
+            "resistance! Correction step #%d.\n",
+            count);
+        count++;
+    }
+    return {R_gg, R_gs};
+}
+
+constexpr std::pair<int, int> BHE_1U::inflow_outflow_bc_component_ids[];
+
+void BHE_1U::updateHeatTransferCoefficients(double const flow_rate)
+
+{
+    _flow_velocity = flow_rate / _pipes.inlet.area();
+
+    double const Re = reynoldsNumber(std::abs(_flow_velocity),
+                                     _pipes.inlet.diameter,
+                                     refrigerant.dynamic_viscosity,
+                                     refrigerant.density);
+    double const Pr = prandtlNumber(refrigerant.dynamic_viscosity,
+                                    refrigerant.specific_heat_capacity,
+                                    refrigerant.thermal_conductivity);
+
+    double const Nu =
+        nusseltNumber(Re, Pr, _pipes.inlet.diameter, borehole_geometry.length);
+
+    _thermal_resistances = calcThermalResistances(Nu);
+}
+
+/// Nu is the Nusselt number.
+std::array<double, BHE_1U::number_of_unknowns> BHE_1U::calcThermalResistances(
+    double const Nu)
+{
+    static constexpr double pi = boost::math::constants::pi<double>();
+
+    double const& lambda_r = refrigerant.thermal_conductivity;
+    double const& lambda_g = grout.lambda_g;
+    double const& lambda_p = _pipes.inlet.wall_thermal_conductivity;
+
+    // thermal resistances due to advective flow of refrigerant in the _pipes
+    // Eq. 36 in Diersch_2011_CG
+    double const R_adv_i1 = 1.0 / (Nu * lambda_r * pi);
+    double const R_adv_o1 = 1.0 / (Nu * lambda_r * pi);
+
+    // thermal resistance due to thermal conductivity of the pipe wall material
+    // Eq. 49
+    double const R_con_a =
+        std::log(_pipes.outlet.diameter / _pipes.inlet.diameter) /
+        (2.0 * pi * lambda_p);
+
+    // the average outer diameter of the _pipes
+    double const& d0 = _pipes.outlet.diameter;
+    double const& D = borehole_geometry.diameter;
+    // Eq. 51
+    double const chi = std::log(std::sqrt(D * D + 2 * d0 * d0) / 2 / d0) /
+                       std::log(D / std::sqrt(2) / d0);
+    // Eq. 52
+    // thermal resistances of the grout
+    double const R_g =
+        std::acosh((D * D + d0 * d0 - _pipes.distance * _pipes.distance) /
+                   (2 * D * d0)) /
+        (2 * pi * lambda_g) * (1.601 - 0.888 * _pipes.distance / D);
+    // thermal resistance due to the grout transition.
+    double const R_con_b = chi * R_g;
+    // Eq. 29 and 30
+    double const R_fig = R_adv_i1 + R_con_a + R_con_b;
+    double const R_fog = R_adv_o1 + R_con_a + R_con_b;
+
+    // thermal resistance due to inter-grout exchange
+    double const R_ar =
+        std::acosh((2.0 * _pipes.distance * _pipes.distance - d0 * d0) / d0 /
+                   d0) /
+        (2.0 * pi * lambda_g);
+
+    double R_gg, R_gs;
+    std::tie(R_gg, R_gs) = thermalResistancesGroutSoil(chi, R_ar, R_g);
+
+    return {R_fig, R_fog, R_gg, R_gs};
```

Comment:
Compilation warning:
```
BHE_1U.cpp:234:13: warning: suggest braces around initialization of subobject [-Wmissing-braces]
    return {R_fig, R_fog, R_gg, R_gs};
            ^~~~~~~~~~~~~~~~~~~~~~~~
            {                       }
```
Please change it to 
```
return {{R_fig, R_fog, R_gg, R_gs}};
```

Reply:
Fixed in https://github.com/ufz/ogs/commit/68af0071714b7be6abf95c880c959e9fa76304eb